src/httpProblemDetail.ts

```diff
@@ -0,0 +1,38 @@
+import { HttpProblemDetailDefinition } from './httpProblemDetailDefinition';
+import { URI, HttpStatusCode } from './types';
+
+const forbiddenDetailNames = ['type', 'status', 'title'];
+
+const defaultType = 'about:blank';
+
+export class HttpProblemDetail {
+  public readonly type: URI;
+  public readonly title: string;
+  public readonly status: HttpStatusCode;
+  private additionalDetails: Record<string, any>;
+
+  constructor({ type, title, status }: HttpProblemDetailDefinition) {
+    this.type = type || defaultType;
+    this.title = title;
+    this.status = status;
+    this.additionalDetails = {};
+  }
+
+  public addAdditionalDetail(key: string, value: any) {
+    if (forbiddenDetailNames.includes(key)) {
+      throw new Error(`${key} is forbidden details name`);
+    }
+    if (value) {
+      this.additionalDetails[key] = value;
+    }
+  }
+
+  toJson(): Record<string, any> {
```

Comment:
**[could]** I would add rename this method to `toPlainObject` and add method:
```
toJson(): JsonString {
  return JSON.stringify(this.toPlainObject())
}
```

This library should be meant to be used in many apps/libs/frameworks. I guess some would prefer to get just plain object other `JSON` in understanding as a human-readable text.